pkg/helm/http_renderer.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+
+	"istio.io/pkg/log"
+)
+
+// HTTPTemplateRenderer is a helm template renderer for helm charts hosted on http url
+type HTTPTemplateRenderer struct {
+	namespace        string
+	componentName    string
```

Comment:
We need to discuss whether it makes sense to have a HTTPTemplateRenderer vs. doing the fetching externally and passing params to the file renderer. 
The reason is that we'll be releasing a build image with all the charts as a single tar and only one poller should be checking that, not every component.